src/test/java/com/pgs/client/interceptor/AuthenticationInterceptorTest.java

```diff
@@ -0,0 +1,68 @@
+package com.pgs.client.interceptor;
+
+import com.pgs.client.dto.Token;
+import com.pgs.client.supplier.AccessTokenSupplier;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import java.util.Random;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class AuthenticationInterceptorTest {
+
+    @Mock
+    private AccessTokenSupplier accessTokenSupplier;
+    @Mock
+    private HttpRequest httpRequest;
```

Comment:
the test checks if service adds a header to httpRequest, therefore it shouldn't be `@Mock` annotated
spring delivers a class MockClientHttpRequest that could be used here